Allow architecture diagrams to support any character for titles

wizbit opened the pull request:
## :bookmark_tabs: Summary

Allows any character between `[` and `]` for architecture diagram titles.

Resolves #5928 

## :straight_ruler: Design Decisions

Change of regex to allow any character. Also had to tweak the `entry Architecture` rule as tests were claiming ambiguity.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://mermaid.js.org/community/contributing.html)
- [x] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://mermaid.js.org/community/contributing.html#update-documentation) is used for all new features.
- [ ] :butterfly: If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

wizbit committed:
Allow architecture diagrams to support any character for titles

Architecture diagrams with Markdown

sidharthv96 commented on packages/parser/tests/architecture.test.ts:
```diff
@@ -0,0 +1,68 @@
+    expect(result.value.$type).toBe(Architecture);
+  });
+
+  it.each([
+    `architecture-beta group api(cloud)[API]`,
+    `  architecture-beta  group api(cloud)[API]  `,
+    `\tarchitecture-beta\tgroup api(cloud)[API]\t`,
+    `
+    architecture-beta\tgroup api(cloud)[API]
+    `,
+  ])('should handle architecture-beta & group in same line', (context: string) => {
+    const result = parse(context);
+    expectNoErrorsOrAlternatives(result);
+    expect(result.value.$type).toBe(Architecture);
```
We should **not** be doing this. Can you change the grammar to disallow this?

sidharthv96 reviewed (changes requested):
Allowing all characters seems okay at the first glance. Love the unit tests!
We should unify the string handling approach, and support markdown as well. (We already support it in flowcharts and some others, so the same functions should be used).

---

We should also make the grammar stricter to disallow `architecture-beta` and `group` in the same line. As the grammar is in beta, we can make breaking changes. 

---

After this is finalized, we also need E2E tests and documentation updates, which show the capabilities in examples.

NicolasNewman commented:
Good catch on the ambiguous grammar! If markdown support is planning on being added we may also want to consider KaTeX support too. Other then the above request changes everything looks good to me.

wizbit commented:
Added markdown support. It already had the ability to render, so added grammar rules for ```["`...`"]```and `["..."]`.
KaTeX support is tricker. I've added it to the demos but it does not render correctly. Turning on `htmlLabels` for rendering makes it work but breaks everything else, so not sure how to proceed with KaTeX.

wizbit commented:
I have discovered that any text that looks like an email does not render correctly in markdown.
I have a fix, but I cannot commit it as the pre-commit linter states it cannot read `packages/mermaid/src/rendering-util/handle-markdown-text.ts`, file being changed is `packages/mermaid/src/rendering-util/handle-markdown-text.ts`

How to I fix that?
